src/apps/api/views/leaderboards.py

```diff
@@ -43,15 +44,24 @@ def update(self, request, *args, **kwargs):
         return response
 
 
+def is_user_submission_owner_or_admin(request, submission, competition):
+    if request.user.is_superuser or request.user in competition.collaborators.all():
```

Comment:
```suggestion
    if request.user.is_superuser or competition.collaborators.filter(pk=request.user.pk).exists():
```
This will keep Django from making a request for all of the collaborators, and instead just tell us if there is one.

Reply:
Changed the query, but left it at the end of the `or` chain so it won't have to run every time.